ui/app/components/job-editor.js

```diff
@@ -95,4 +122,43 @@ export default class JobEditor extends Component {
     const [file] = event.target.files;
     reader.readAsText(file);
   }
+
+  @action
+  toggleView() {
+    const opposite = this.view === 'job-spec' ? 'full-definition' : 'job-spec';
+    this.view = opposite;
+  }
+
+  get definition() {
+    if (this.view === 'full-definition') {
+      return this.args.definition;
+    } else {
+      return this.args.specification;
+    }
+  }
+
+  get data() {
+    return {
+      cancelable: this.args.cancelable,
+      definition: this.definition,
+      job: this.args.job,
+      planOutput: this.planOutput,
+      shouldShowPlanMessage: this.shouldShowPlanMessage,
+      view: this.view,
+    };
+  }
+
+  get fns() {
+    return {
+      onCancel: this.onCancel,
+      onEdit: this.edit,
+      onPlan: this.plan,
+      onReset: this.reset,
+      onSaveAs: this.args.handleSaveAsTemplate,
+      onSubmit: this.submit,
+      onToggle: this.toggleView,
+      onUpdate: this.updateCode,
+      onUpload: this.uploadJobSpec,
+    };
+  }
```

Comment:
Hey, this is a neat pattern! Is this something we use anyplace else in Nomad?

Reply:
Yes! We use this in `breadcrumbs` and `evaluations`! Any major net new features I've worked on!